nat-lab/tests/conftest.py

```diff
@@ -68,58 +69,94 @@ def event_loop():
+        cmd = [
             "netsh",
             "int",
             "ipv4",
             "set",
             "dynamic",
             "tcp",
-            "start=" + str(start_port),
-            "num=" + str(num_ports),
-        ]).execute(on_output, on_output)
+            f"start={start_port}",
+            f"num={num_ports}",
+        ]
+    elif vm_tag is ConnectionTag.MAC_VM:
+        cmd = [
+            "sysctl",
+            "-w",
+            f"net.inet.ip.portrange.first={start_port}",
+            f"net.inet.ip.portrange.last={start_port + num_ports}",
+        ]
+    else:
+        # Linux
+        cmd = [
+            "sysctl",
+            "-w",
+            f"net.ipv4.ip_local_port_range={start_port} {start_port + num_ports}",
+        ]
 
+    async with new_connection_raw(vm_tag) as connection:
+        await connection.create_process(cmd).execute(on_output, on_output)
 
-@pytest.fixture(autouse=True)
-@pytest.mark.windows
-def setup_windows_ports(request):
-    test_name = request.node.name
 
+@pytest.fixture(autouse=True)
+def setup_ephemeral_ports(request):
     def execute_setup(vm_tag):
         try:
-            asyncio.run(win_ports(vm_tag))
+            asyncio.run(os_ephemeral_ports(vm_tag))
         except ProcessExecError as e:
-            print(f"win_ports_{vm_tag} process execution failed: {e}")
+            print(
+                datetime.now(),
+                f"os_ephemeral_ports_{vm_tag} process execution failed: {e}",
+            )
+
+    connection_tags = set()
+
+    # Setup for all Docker clients
+    docker_clients = [
+        tag
+        for tag in ConnectionTag.__members__.values()
+        if tag.name.startswith("DOCKER_") and "CLIENT" in tag.name
+    ]
+    for client in docker_clients:
+        connection_tags.add(client)
 
-    if "[WINDOWS_VM_" in test_name:
-        # Extract all VM numbers (WINDOWS_VM_1 and/or WINDOWS_VM_2)
+    # Handle test name (params) to search for VM tags
+    test_name = request.node.name
+    if "[" in test_name and "]" in test_name and "_VM_" in test_name:
+        # Extract all connection tags from test name
         vm_names = [
             param
             for param in test_name.split("[")[1].split("]")[0].split("-")
-            if param.startswith("WINDOWS_VM_")
+            if param in ConnectionTag.__members__
         ]
 
         for vm_name in vm_names:
-            execute_setup(ConnectionTag[vm_name])
-    else:
-        items = request.session.items
-        for item in items:
-            for mark in item.own_markers:
-                if mark.name != "windows":
-                    continue
-                for tag in [ConnectionTag.WINDOWS_VM_1, ConnectionTag.WINDOWS_VM_2]:
-                    execute_setup(tag)
+            connection_tags.add(ConnectionTag[vm_name])
 
-                return
+    # Handle test markers to search for VMs tags
+    test_items = request.session.items
```

Comment:
nit: this part is ok, but I'd rewrite it differently, using nested indentations tho.

```python
if any(
    mark.name == "windows"
    for item in request.session.items
    for mark in item.own_markers
):
    tags += [ConnectionTag.WINDOWS_VM_1, ConnectionTag.WINDOWS_VM_2]

if any(
    mark.name == "mac"
    for item in request.session.items
    for mark in item.own_markers
):
    tags += [ConnectionTag.MAC_VM]
```